### PR TITLE
DTFS2-4607 - Facility loan record creation

### PR DESCRIPTION
**Pull request by abhi-markan**
### Introduction
Facility loan records are only created for `issued` facilities with activated bundles. This is a cross-deal and cross-submission type ticket. Following additions and improvements have been achieved.

* Introduction of new durable function responsible for yielding loan record `activity-create-facility-loan`
* Introduction of `addDay` and `addMonth` functions in centralised date helpers collection.
* Addition of new facility level constants
* Improved documentation
* Lint fixes

### CODE REVIEW

File: azure-functions/acbs-function/mappings/facility/facility-loan.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const facilityLoan = (deal, facility, acbsData) => {
+  const issueDate = helpers.getIssueDate(facility, getDealSubmissionDate(deal));
+  const { guaranteeExpiryDate } = facility.tfm.facilityGuaranteeDates
+    ? facility.tfm.facilityGuaranteeDates
+    : '';
+
+  let loanRecord = {
+    portfolioIdentifier: CONSTANTS.FACILITY.PORTFOLIO.E1,
+    postingDate: date.now(),
```
Comment (IgnatG) [inline azure-functions/acbs-function/mappings/facility/facility-loan.js:37]: Does this work? Or do we need to use `Date.now()` ?

**Review by IgnatG — COMMENTED** [review]
Beautiful! - well done 
I've left some small comments. Not show stoppers though

**Review by IgnatG — APPROVED** [review]

File: azure-functions/acbs-function/constants/facility.js [hunk 2]
```diff
@@ -11,6 +11,12 @@ const FACILITY_TYPE_CODE = {
   GEF: '280',
 };
 
+const PRODUCT_TYPE_GROUP = {
+  EWCS: 'EW',
+  BSS: 'BS',
+  GEF: 'GM',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply azure-functions/acbs-function/constants/facility.js:17]: This is a value specific for `GEF` deal type, it is an ACBS code.

File: azure-functions/acbs-function/mappings/facility/facility-loan.js (hunk 1 above)
Comment (abhi-markan) [reply azure-functions/acbs-function/mappings/facility/facility-loan.js:37]: This does it is method from `date.js` helpers collection file.

**Review by IgnatG — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]
Could do with some tests but happy to get this in, great stuff!

**Pull request merged**